From 520a20e731a903690343d8a4d7bbe11ca40e11e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jo=C3=A3o=20Paulo=20Gon=C3=A7alves?=
 <joao.goncalves@toradex.com>
Date: Fri, 18 Apr 2025 12:38:58 -0300
Subject: [PATCH] i2c: omap: Increase timeout to 5 seconds
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When adding the gmsl serializer, the I2C bus is timing out for other
devices on the bus. Increasing the timeout to 5 seconds solved the
poblem.

Don't know why it is happening, maybe the bus is being held for longer than 1
sec to communicate with the serializier over the gmsl link. Need to investigate
more. For now, I will go foward to make the whole system to work.

Upstream-Status: Pending
Signed-off-by: João Paulo Gonçalves <jpaulo.silvagoncalves@gmail.com>
---
 drivers/i2c/busses/i2c-omap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-omap.c b/drivers/i2c/busses/i2c-omap.c
index 82fd45ebcf1fc7..cd1dae2deb3655 100644
--- a/drivers/i2c/busses/i2c-omap.c
+++ b/drivers/i2c/busses/i2c-omap.c
@@ -42,7 +42,7 @@
 #define OMAP_I2C_REV_ON_4430_PLUS	0x50400002
 
 /* timeout waiting for the controller to respond */
-#define OMAP_I2C_TIMEOUT (msecs_to_jiffies(1000))
+#define OMAP_I2C_TIMEOUT (msecs_to_jiffies(5000))
 
 /* timeout for pm runtime autosuspend */
 #define OMAP_I2C_PM_TIMEOUT		1000	/* ms */
